charms/argo-controller/reactive/argo_controller.py

```diff
@@ -32,7 +32,15 @@ def start_charm():
 
     layer.status.maintenance('configuring container')
 
-    minio = endpoint_from_name('minio').services()[0]['hosts'][0]
+    try:
+        minio = endpoint_from_name('minio').mailman3()[0]
+    except IndexError:
+        layer.status.blocked('Waiting for minio relation.')
```

Comment:
Suggest changing this to `.waiting` instead of `.blocked` -- you have `endpoint.minio.joined`, so there's nothing for an operator to do but wait.